sql: TestDropTableWhileUpgradingFormat failed

sql.TestDropTableWhileUpgradingFormat [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11227148?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11227148?buildTab=artifacts#/) on master @ [8c1ff5188f37d8b3b4b1f153c16a38e876e53973](https://github.com/cockroachdb/cockroach/commits/8c1ff5188f37d8b3b4b1f153c16a38e876e53973):


```
=== RUN   TestDropTableWhileUpgradingFormat
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/765430912bd62afa1f42a127d71d4b8a/logTestDropTableWhileUpgradingFormat3691027404
    test_log_scope.go:81: use -show-logs to present logs inline
    drop_test.go:828: expecting 100 keys, but found 0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/765430912bd62afa1f42a127d71d4b8a/logTestDropTableWhileUpgradingFormat3691027404
--- FAIL: TestDropTableWhileUpgradingFormat (2.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropTableWhileUpgradingFormat.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30431

sql.TestDropTableWhileUpgradingFormat [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11286710?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11286710?buildTab=artifacts#/) on master @ [a629092bb552144ec05954ad8c000e35c8da8ef0](https://github.com/cockroachdb/cockroach/commits/a629092bb552144ec05954ad8c000e35c8da8ef0):


```
=== RUN   TestDropTableWhileUpgradingFormat
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/1061c85d1b78da61f97933da6fe84ebc/logTestDropTableWhileUpgradingFormat2788016457
    test_log_scope.go:81: use -show-logs to present logs inline
    drop_test.go:828: expecting 100 keys, but found 0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1061c85d1b78da61f97933da6fe84ebc/logTestDropTableWhileUpgradingFormat2788016457
--- FAIL: TestDropTableWhileUpgradingFormat (1.62s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropTableWhileUpgradingFormat.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql.TestDropTableWhileUpgradingFormat [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11353133?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11353133?buildTab=artifacts#/) on master @ [6716e92795a8362a7b49cb33443d1b07e536f325](https://github.com/cockroachdb/cockroach/commits/6716e92795a8362a7b49cb33443d1b07e536f325):


```
=== RUN   TestDropTableWhileUpgradingFormat
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/230b14ff9b70e2729ccc9b9deabcc166/logTestDropTableWhileUpgradingFormat875673394
    test_log_scope.go:81: use -show-logs to present logs inline
    drop_test.go:832: expecting 100 keys, but found 0
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/230b14ff9b70e2729ccc9b9deabcc166/logTestDropTableWhileUpgradingFormat875673394
--- FAIL: TestDropTableWhileUpgradingFormat (0.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDropTableWhileUpgradingFormat.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/drop_test.go
+++ b/pkg/sql/drop_test.go
@@ -829,7 +829,12 @@ func TestDropTableWhileUpgradingFormat(t *testing.T) {
 	}
 
 	tableSpan := tableDesc.TableSpan(keys.SystemSQLCodec)
-	tests.CheckKeyCountIncludingTombstoned(t, s, tableSpan, numRows)
+	testutils.SucceedsSoon(t, func() error {
+		if err := tests.CheckKeyCountIncludingTombstonedE(t, s, tableSpan, numRows); err != nil {
+			return errors.Wrap(err, "failed to verify expected amount of keys")
+		}
+		return nil
+	})
 
 	sqlDB.Exec(t, `DROP TABLE test.t`)
 

--- a/pkg/sql/tests/data.go
+++ b/pkg/sql/tests/data.go
@@ -45,7 +45,7 @@ func CheckKeyCountIncludingTombstoned(
 	}
 }
 
-// CheckKeyCountE returns an error if the the number of keys in the
+// CheckKeyCountE returns an error if the number of keys in the
 // provided span does not match numKeys.
 func CheckKeyCountE(t *testing.T, kvDB *kv.DB, span roachpb.Span, numKeys int) error {
 	t.Helper()

